src/number/n_transExt.jl

```diff
@@ -0,0 +1,360 @@
+
+elem_type(::Type{N_FField}) = n_transExt
+
+parent(a::n_transExt) = a.parent
+
+parent_type(::Type{n_transExt}) = N_FField
+
+base_ring(a::n_transExt) = Union{}
+
+base_ring(a::N_FField) = a.base_ring
+
+@doc Markdown.doc"""
+    transcendence_degree(F::N_FField)
+> Return the trancendence degree of the given function field.
+"""
+transcendence_degree(F::N_FField) = Int(libSingular.rPar(F.ptr))
+
+@doc Markdown.doc"""
+    basis(F::N_FField)
+> Return the trancendence basis of the given function field.
+"""
+function transcendence_basis(F::N_FField)
+   n = transcendence_degree(F)
+   return [F(libSingular.n_Param(Cint(i), F.ptr)) for i = 1:n]
+end
+
+@doc Markdown.doc"""
+   characteristic(R::N_FField)
+> Return the characteristic of the field.
+"""
+function characteristic(R::N_FField)
+   return ZZ(libSingular.n_GetChar(R.ptr))
+end
+
+function deepcopy_internal(a::n_transExt, dict::IdDict)
+   return parent(a)(libSingular.n_Copy(a.ptr, parent(a).ptr))
+end
+
+#TODO: rewrite hash function
+function hash(a::n_transExt, h::UInt)
+   return 0x2c42e12d0c837511%UInt
+end
+
+###############################################################################
+#
+#   Basic manipulation
+#
+###############################################################################
+
+one(R::N_FField) = R(1)
+
+zero(R::N_FField) = R(0)
+
+@doc Markdown.doc"""
+    numerator(n::n_transExt)
+> Return the numerator of the given fraction.
+"""
+function numerator(n::n_transExt)
+   F = parent(n)
+   return F(libSingular.n_GetNumerator(n.ptr, F.ptr))
+end
+
+@doc Markdown.doc"""
+    denominator(n::n_transExt)
+> Return the denominator of the given fraction.
+"""
+function denominator(n::n_transExt)
+   F = parent(n)
+   return F(libSingular.n_GetDenom(n.ptr, F.ptr))
+end
+
+function isone(n::n_transExt)
+   c = parent(n)
+   return libSingular.n_IsOne(n.ptr, c.ptr)
+end
+
+function iszero(n::n_transExt)
+   c = parent(n)
+   return libSingular.n_IsZero(n.ptr, c.ptr)
+end
+
+@doc Markdown.doc"""
+   isunit(n::n_transExt)
+> Return `true` if $n$ is a unit in the field, i.e. nonzero.
+"""
+isunit(n::n_transExt) = !iszero(n)
+
+###############################################################################
+#
+#   Canonicalisation
+#
+###############################################################################
+
+canonical_unit(x::n_transExt) = x
+
+###############################################################################
+#
+#   String I/O
+#
+###############################################################################
+
+function show(io::IO, F::N_FField)
+   print(io, "Function Field over ", base_ring(F), " with transcendence basis ", transcendence_basis(F))
+end
+
+function show(io::IO, n::n_transExt)
+   libSingular.StringSetS("")
+
+   libSingular.n_Write(n.ptr, parent(n).ptr, false)
+
+   m = libSingular.StringEndS()
+   
+   print(io, m)
+end
+
+needs_parentheses(x::n_transExt) = false
+
+isnegative(x::n_transExt) = x == -1
```

Comment:
Is this really implemented correctly? Take a look at the AbstractAlgebra documentation to see.

Reply:
This was a copy/paste error I did. The function does not make sense for function fields.